.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,18 @@
+# Azure IoTEdge PR checklist:
+
+This checklist is used to make sure that common guidelines for a pull request are followed.
+
+### General Guidelines and Best Practices
+- [ ] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
+- [ ] Title of the pull request is clear and informative.
+- [ ] Description of the pull request includes a brief summary of the enhancement or bug fix.
+- [ ] There are a small number of commits, each with an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
```

Comment:
I don't fully understand this one. I think the statement about previously merged commits showing up in the PR probably goes without saying. And I think the powershell doc focuses on the wrong things, like # of commits (vs. # of concepts in the PR) and details like rebase vs. merge.

I do think PRs should be scoped to focus on a single thing (e.g., fixing a bug _and_ doing a codebase-wide reformatting of braces in one PR == bad), and split into multiple PRs as necessary.

Reply:
Also, I like the steps that kodiak takes when we add the 'ready-to-merge' label to a PR:
- PR title becomes commit title
- PR description becomes commit description
- PR is squashed and merged (so the # of commits in the PR, and their individual messages, don't really matter)

I don't think we have to mandate the use of that label, but whether you use it or not the merge commit should look more or less the same.

Reply:
Interesting. Given that kodiak makes the whole PR into a single commit, lets reference that in our PR template.